core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -859,9 +857,10 @@ public void forceMerge(boolean flush) throws IOException {
      * Snapshots the index and returns a handle to it. If needed will try and "commit" the
      * lucene index to make sure we have a "fresh" copy of the files to snapshot.
      *
+     * @param safeCommit indicates whether the engine should acquire the most recent safe commit, or the most recent commit.
      * @param flushFirst indicates whether the engine should flush before returning the snapshot
      */
-    public abstract IndexCommitRef acquireIndexCommit(boolean flushFirst) throws EngineException;
+    public abstract IndexCommitRef acquireIndexCommit(boolean safeCommit, boolean flushFirst) throws EngineException;
```

Comment:
we should look in future to remove one of these booleans - either you want to "get everything" or you want a safe commit. I don't think there's a point in "flush but give me a safe commit"

Reply:
I agree. Are you ok if we replace this by having two methods: `acquireLastIndexCommit(flushFirst)` and `acquireSafeIndexCommit(no option)`?

Reply:
>  agree. Are you ok if we replace this by having two methods: acquireLastIndexCommit(flushFirst) and acquireSafeIndexCommit(no option)?

+1

Reply:
double checking - the new methods will be a follow up?